### PR TITLE
Implements user search

### PR DESCRIPTION
**Pull request by maxwang7**
Associated worksheets PR: codalab-worksheets:#379: https://github.com/codalab/codalab-worksheets/pull/379

### CODE REVIEW

File: codalab/model/bundle_model.py [hunk 1]
```diff
@@ -1726,6 +1726,41 @@ def get_user(self, user_id=None, username=None, check_active=True):
         if result:
             return result[0]
         return None
+    
+    def search_users(self, user_id, keywords):
+        clauses = []
+        limit = 5
+        for keyword in keywords:
+            if '.limit' not in keyword:
```
Comment (maxwang7) [inline codalab/model/bundle_model.py:1734]: I'm not sure the best way to filter out filters (e.g. `.limit` or `.sort=reverse`). Thoughts?

File: codalab/model/bundle_model.py [hunk 2]
```diff
@@ -1726,6 +1726,41 @@ def get_user(self, user_id=None, username=None, check_active=True):
...
+            cl_user.c.is_active,
+            cl_user.c.first_name,
+            cl_user.c.last_name,
+            cl_user.c.affiliation,
+            cl_user.c.time_quota,
+            cl_user.c.time_used,
+            cl_user.c.disk_quota,
+            cl_user.c.disk_used,
+            cl_user.c.disk_quota,
+        ]
```
Comment (maxwang7) [inline codalab/model/bundle_model.py:1748]: Are there fields we would want to include/exclude?

Comment (percyliang) [reply codalab/model/bundle_model.py:1748]: Don't include sensitive information (e.g., email)!  I'd only include information that's going to show up on the webpage.  Err on the side of including less.

File: codalab/rest/users.py [hunk 3]
```diff
@@ -76,20 +79,27 @@ def fetch_user(user_spec):
...
-    # Combine username and email filters
-    usernames = set(request.query.get('filter[user_name]', '').split(','))
-    usernames |= set(request.query.get('filter[email]', '').split(','))
-    usernames.discard('')  # str.split(',') will return '' on empty strings
-    users = local.model.get_users(usernames=(usernames or None))
-    return allowed_user_schema()(many=True).dump(users).data
+    keywords = query_get_list('keywords')
+    if keywords:
+        users = local.model.search_users(request.user.user_id, keywords)
+        return allowed_user_schema()(many=True).dump(users).data
```
Comment (percyliang) [inline codalab/rest/users.py:95]: Take this line out of the if/else to avoid duplication

File: codalab/model/bundle_model.py [hunk 4]
```diff
@@ -1726,6 +1726,41 @@ def get_user(self, user_id=None, username=None, check_active=True):
...
+        query = select(cols_to_select).distinct().where(clause).limit(limit)
+        with self.engine.begin() as connection:
+            rows = connection.execute(query).fetchall()
+            if not rows:
+                return []
```
Comment (percyliang) [inline codalab/model/bundle_model.py:1753]: Does this return `None`?  Can't return rows directly?

File: codalab/model/bundle_model.py (hunk 1 above)
Comment (percyliang) [reply codalab/model/bundle_model.py:1734]: A hack is to just ignore anything that starts with a period.  Definitely should make this more generic.  If we're not going to respect `limit`, then we should document that.

**Review by percyliang — CHANGES_REQUESTED** [review]
Mostly looks good, just a few localized changes to make.

**Pull request closed without merging**